### PR TITLE
[SR-14824] Improve diagnostic for multi-statement closures instead of…

### PR DESCRIPTION
**Pull request by JiarenWang**
[SR-14824] Improve diagnostic for multi-statement closures instead of saying "too complex closure return type"

Resolves SR-14824.

### CODE REVIEW

**Comment by JiarenWang** [comment]
@varungandhi-apple Pelease review.

File: include/swift/AST/DiagnosticsSema.def [hunk 1]
```diff
@@ -273,7 +273,7 @@ ERROR(cannot_infer_closure_parameter_type,none,
 ERROR(cannot_infer_closure_type,none,
       "unable to infer closure type in the current context", ())
 ERROR(cannot_infer_closure_result_type,none,
-      "unable to infer%select{ complex|}0 closure return type; "
+      "unable to infer multiple-statement closure return type; "
       "add explicit type to disambiguate", (bool))
```
Comment (varungandhi-apple) [inline include/swift/AST/DiagnosticsSema.def:277]: Based on some discussion with other folks, it looks like the single-expression case (when the boolean is true) cannot be hit at all. So you could remove the `(bool)` from the below line and update the place in `CSDiagnostics.cpp` emitting the diagnostic accordingly.

Wording-wise, perhaps it would be better to rephrase this to:

```suggestion
      "cannot infer return type for closure with multiple statements; "
      "add explicit type to disambiguate", ())
```

**Review by varungandhi-apple — CHANGES_REQUESTED** [review]

File: include/swift/AST/DiagnosticsSema.def (hunk 1 above)
Comment (xedin) [reply include/swift/AST/DiagnosticsSema.def:277]: Yes, I think that single-statement cloure check might just be an artifact from the CSDiag days which is no longer useful. Also I like the phrasing @varungandhi-apple suggested.

**Comment by JiarenWang** [comment]
@varungandhi-apple Pelease review.

**Review by varungandhi-apple — APPROVED** [review]
Thanks, looks good to me. 👍🏼

**Comment by varungandhi-apple** [comment]
@swift-ci smoke test and merge

**Review by xedin — APPROVED** [review]
Thank you!

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test MacOS Platform

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please test Windows Platform
